src/margo-init.c

```diff
@@ -1438,16 +1472,70 @@ static int create_xstream_from_config(struct json_object*          es_config,
     return ABT_SUCCESS;
 }
 
-static void set_argobots_environment_variables(struct json_object* config)
+static void confirm_argobots_configuration(struct json_object* config)
 {
+    /* this function assumes that the json is already fully populated */
+    size_t runtime_abt_thread_stacksize = 0;
+
+    /* retrieve expected values according to Margo configuration */
     struct json_object* argobots = json_object_object_get(config, "argobots");
-    int                 abt_mem_max_num_stacks = json_object_get_int64(
-        json_object_object_get(argobots, "abt_mem_max_num_stacks"));
-    int abt_thread_stacksize = json_object_get_int64(
+    int                 abt_thread_stacksize = json_object_get_int64(
         json_object_object_get(argobots, "abt_thread_stacksize"));
 
+    /* NOTE: we skip checking num_stacks; this cannot be retrieved with
+     * ABT_info_query_config(). Fortunately it also is not as crucial as the
+     * stack size.  Recent ABT releases have conservative caps on stack
+     * cache sizes by default.
+     */
+
+    /* query Argobots to see if it is in agreement */
+    ABT_info_query_config(ABT_INFO_QUERY_KIND_DEFAULT_THREAD_STACKSIZE,
+                          &runtime_abt_thread_stacksize);
+    if (runtime_abt_thread_stacksize != abt_thread_stacksize) {
+        MARGO_WARNING(0,
+                      "Margo requested an Argobots ULT stack size of %d, but "
+                      "Argobots is using a ULT stack size of %zd.",
+                      abt_thread_stacksize, runtime_abt_thread_stacksize);
+        MARGO_WARNING(
```

Comment:
I think the only thing I would change is make those three MARGO_WARNING a single (long) one, just so it would appear as a single line in a log file produced by multiple processes.

Reply:
That makes sense. I just pushed the change.